src/ir.cpp

```diff
@@ -15669,6 +15738,323 @@ static TypeTableEntry *ir_analyze_instruction_offset_of(IrAnalyze *ira,
     return ira->codegen->builtin_types.entry_num_lit_int;
 }
 
+static ConstExprValue *ir_make_type_info_value(IrAnalyze *ira, ConstExprValue *parent, TypeTableEntry *type_entry)
+{
+    assert(type_entry != nullptr);
+    assert(!type_is_invalid(type_entry));
+
+    switch (type_entry->id)
+    {
+        case TypeTableEntryIdInvalid:
+            zig_unreachable();
+        case TypeTableEntryIdMetaType:
+        case TypeTableEntryIdVoid:
+        case TypeTableEntryIdBool:
+        case TypeTableEntryIdUnreachable:
+        case TypeTableEntryIdNumLitFloat:
+        case TypeTableEntryIdNumLitInt:
+        case TypeTableEntryIdUndefLit:
+        case TypeTableEntryIdNullLit:
+        case TypeTableEntryIdNamespace:
+        case TypeTableEntryIdBlock:
+        case TypeTableEntryIdArgTuple:
+        case TypeTableEntryIdOpaque:
+            return nullptr;
+        case TypeTableEntryIdInt:
+            {
+                ConstExprValue *payload = create_const_vals(1);
+                payload->special = ConstValSpecialStatic;
+
+                ConstExprValue *int_info_type = get_builtin_value(ira->codegen, "IntInfo");
+                assert(int_info_type->type->id == TypeTableEntryIdMetaType);
+                payload->type = int_info_type->data.x_type;
+
+                ConstExprValue *fields = create_const_vals(2);
+                payload->data.x_struct.fields = fields;
+
+                if (parent->type->id == TypeTableEntryIdStruct)
+                {
+                    payload->data.x_struct.parent.id = ConstParentIdStruct;
+                    payload->data.x_struct.parent.data.p_union.union_val = parent;
+                }
+                else if (parent->type->id == TypeTableEntryIdUnion)
+                {
+                    payload->data.x_struct.parent.id = ConstParentIdUnion;
+                    payload->data.x_struct.parent.data.p_union.union_val = parent;
+                }
+                else
+                {
+                    payload->data.x_struct.parent.id = ConstParentIdNone;
+                }
+
+                // is_signed: bool
+                fields[0].special = ConstValSpecialStatic;
```

Comment:
unfortunate that we have these footguns here. (this problem exists elsewhere in ir.cpp, so you're not introducing a new problem). but maybe we could add some asserts like this?
```c++
assert(find_struct_type_field(int_info_type->data.x_type, buf_create_from_str("is_signed"))->src_index == 0);
```
so that we at least get asserts in debug builds if the code in codegen.cpp that writes builtin.zig is changed. and then in release mode we don't do the find.